spec/cohort_manager.spec.js

```diff
@@ -0,0 +1,290 @@
+    manager.createCohort('Cohort2')
+    manager.createCohort('Cohort3')
+    result = manager.removeCohort('Cohort2')
+    // verify
+    expect(result).toEqual(updatedList)
+  })
+
+  //TEST 8
+  it('remove a cohort by name - COHORT IS NOT VALID', () => {
+    // setup
+    const manager = new CohortManager()
+    const errorMessage = 'NOT A VALID COHORT NAME'
+    // execute
+    manager.createCohort('Cohort1')
+    manager.createCohort('Cohort2')
+    manager.createCohort('Cohort3')
+    result = manager.removeCohort('Non-Existent Cohort')
+    // verify
+    expect(result).toEqual(errorMessage)
+  })
+
+  //TEST 9
+  it('remove a student from a specific cohort', () => {
+    // setup
+    const manager = new CohortManager()
+    // execute
+    manager.createCohort('Cohort1')
+    manager.createCohort('Cohort2')
+    manager.createCohort('Cohort3')
+    manager.addStudent(
+      'Cohort3',
+      'Mike Wazowski',
+      'mike_GH',
+      'mike@monster.inc'
+    )
+    manager.addStudent(
+      'Cohort3',
+      'James Sullivan',
+      'sullyGH',
+      'sully@sully.inc'
+    )
+    manager.addStudent('Cohort3', 'Randall Boggs', 'boggsGH', 'boggs@inc.com')
+    result = manager.removeStudent('Cohort3', 'Randall')
+    // verify
+    expect(result.students.length).toEqual(2)
+    expect(result.students[0].firstName).toEqual('Mike')
+    expect(result.students[1].firstName).toEqual('James')
+  })
+
+  // TEST 10
+  it('remove a student from a specific cohort - COHORT OR STUDENT NOT FOUND', () => {
+    // setup
+    const manager = new CohortManager()
+    const errorMessage = 'COHORT NOT FOUND'
+    // execute
+    manager.createCohort('Cohort1')
+    manager.createCohort('Cohort2')
+    manager.createCohort('Cohort3')
+    manager.addStudent(
+      'Cohort3',
+      'Mike Wazowski',
+      'mike_GH',
+      'mike@monster.inc'
+    )
+    manager.addStudent(
+      'Cohort3',
+      'James Sullivan',
+      'sullyGH',
+      'sully@sully.inc'
+    )
+    manager.addStudent('Cohort3', 'Randall Boggs', 'boggsGH', 'boggs@inc.com')
+    result = manager.removeStudent('Cohort5', 'Duck')
+    // verify
+    expect(result).toEqual(errorMessage)
+  })
+
+  // TEST 11
+  it('find cohorts with students', () => {
+    // setup
+    const manager = new CohortManager()
+    // execute
+    manager.createCohort('Cohort1')
+    manager.createCohort('Cohort2')
+    manager.createCohort('Cohort3')
+    manager.addStudent(
+      'Cohort3',
+      'Mike Wazowski',
+      'mike_GH',
+      'mike@monster.inc'
+    )
+    manager.addStudent(
+      'Cohort3',
+      'James Sullivan',
+      'sullyGH',
+      'sully@sully.inc'
+    )
+    manager.addStudent('Cohort3', 'Randall Boggs', 'boggsGH', 'boggs@inc.com')
+    result = manager.findCohortsWithStudents()
+    // verify
+    expect(result.length).toEqual(1)
+    expect(result[0].name).toEqual('Cohort3')
+    expect(result[0].students.length).toEqual(3)
+  })
+
+  //TEST 12
```

Comment:
<img width="386" alt="Screenshot 2022-04-04 at 10 02 07" src="https://user-images.githubusercontent.com/98287394/161510732-e3222413-c4eb-4c33-8b62-650c8d672a36.png">
This is the test I wrote for finding a student via ID, again seems to be a lot simpler than yours, am I approaching this wrong or are your methods just more complex etc?